File: script/python/http_util/http_server.py

import Http
import game.util.logger


class HttpServer(object):

    def __init__(self, server_port):
        # print(server_ip, server_port)
        self._http_server = Http.HttpServer(server_port, self)
        assert(self._http_server is not None)

    def on_recv_http_req(self, conn_id, req):
        game.util.logger.log_info("recv http req")
        print(conn_id, req)

    def send_http_resp(self, conn_id, resp):
        self._http_server.sendHttpResp(conn_id, resp)

File: script/lua/base/http_server.lua
require("base.object")
require("util.logger")
require("util.str_util")

clsHttpServer = clsObject:Inherit("clsHttpServer")

function clsHttpServer:__init__(port)
    self._http_server = HttpServer.createHttpServer(port, self)
    logger.logInfo("create http server, %d", port)
end

function clsHttpServer:on_recv_http_req(conn_id, req)
    logger.logInfo("*****recv http req, %d, %s", conn_id, StrUtil.tableToStr(req))
    return {status=200, body="ok"}
end